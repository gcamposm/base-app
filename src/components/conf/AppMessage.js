import { Alert } from 'antd';
import moment from 'moment';

const AppMessage = () => {
  const handleClose = () => {
    const timestamp = moment().unix();
    window.localStorage.setItem('appMessage', timestamp);
  };

  const showMessage = () => {
    const time = window.localStorage.getItem('appMessage');
    if (!time) return true;

    const timestamp = parseInt(time, 10);

    const timeA = moment.unix(parseInt(timestamp, 10));
    const timeB = moment();

    const diff = timeB.diff(timeA, 'hours');

    return diff > 5;
  };

  const description = () => (
    <div>
      El cierre de entregas en mucha comunas al sur de Chile durante el paro de camioneros ha
      generado una acumulación de envíos pendientes y muchas comunas todavía se encuentran cerradas,
      así que no podrás generar tracking o imprimir etiquetas. Te aconsejamos aún no cargar nuevos
      pedidos hacia esas zonas hasta que la situación se estabilice.{' '}
      <a
        onClick={() => {
          window.open('https://shipitcl.zendesk.com/hc/es-419/articles/360053023954');
        }}
        href="#"
      >
        Revisa el detalle aquí
      </a>
      .
    </div>
  );

  return (
    <>
      {showMessage() ? (
        <Alert
          banner
          style={{ margin: '20px' }}
          message="Atención"
          description={description()}
          type="warning"
          showIcon
          closable
          onClose={handleClose}
        />
      ) : null}
    </>
  );
};

export default AppMessage;
