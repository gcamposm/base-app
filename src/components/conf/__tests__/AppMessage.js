import { fireEvent, screen, render } from '@testing-library/react';

import AppMessage from '../AppMessage';

describe('App Message', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'localStorage', {
      value: {
        getItem: jest.fn(() => null),
        setItem: jest.fn(() => null)
      },
      writable: true
    });

    render(<AppMessage />);
  });

  it('should have warning text', () => {
    const text = screen.getByText(/El cierre de entregas en mucha comunas/);
    expect(text).toMatchSnapshot();
  });

  it('should set localStorage variables', async () => {
    expect(window.localStorage.getItem).toHaveBeenCalledTimes(1);
    expect(window.localStorage.setItem).toHaveBeenCalledTimes(0);

    const closeButton = screen.getByRole('button');
    fireEvent.click(closeButton);
    expect(window.localStorage.setItem).toHaveBeenCalledTimes(1);
  });
});
