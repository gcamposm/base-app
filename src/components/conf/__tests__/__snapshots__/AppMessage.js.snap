// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Message should have warning text 1`] = `
<div>
  El cierre de entregas en mucha comunas al sur de Chile durante el paro de camioneros ha generado una acumulación de envíos pendientes y muchas comunas todavía se encuentran cerradas, así que no podrás generar tracking o imprimir etiquetas. Te aconsejamos aún no cargar nuevos pedidos hacia esas zonas hasta que la situación se estabilice.
   
  <a
    href="#"
  >
    Revisa el detalle aquí
  </a>
  .
</div>
`;
